src/applications/apps.js

```diff
@@ -41,61 +42,21 @@ export function getAppStatus(appName) {
   return app ? app.status : null;
 }
 
-export function registerApplication(
-  appName,
-  applicationOrLoadingFn,
-  activityFn,
-  customProps = {}
-) {
-  if (typeof appName !== "string" || appName.length === 0)
-    throw Error(
-      formatErrorMessage(
-        20,
-        __DEV__ &&
-          `The first argument to registerApplication must be a non-empty string 'appName'`
-      )
-    );
-  if (getAppNames().indexOf(appName) !== -1)
-    throw Error(
-      formatErrorMessage(
-        21,
-        __DEV__ && `There is already an app declared with name ${appName}`,
-        appName
-      )
-    );
-  if (typeof customProps !== "object" || Array.isArray(customProps))
-    throw Error(
-      formatErrorMessage(22, __DEV__ && "customProps must be an object")
-    );
+export function registerApplication(...args) {
+  const { name, loadImpl, activityFn, customProps } = sanitizeAPI(args);
```

Comment:
I agree, the option 1 would definitely be better for a project that should be as slim as possible, I will try to keep that in mind next time!
My thought process was that naming `appName` to `appNameOrConfig` would make things go a little bit out of hand, and to avoid clutter inside the `registerApplication` function, I moved all arguments to the "sanitize" (where they are being validated and "transformed"  to correct internal api types, well almost hehehe)
If possible I'd like to keep the flow:
`registerApplication` -> `sanitize()` (which will validate all arguments, regarding form, and map outside arguments to correct internal representation) -> apps.push()